Fix: Очистка каталогов результатов allure перед формированием результатов

Kyrales opened the pull request:
Исправлена ошибка, когда не были очищены старые результат allure в workspace при повторных запусках пайплайнов.

Kyrales committed:
Добавлена очистка каталога Allure при инициализации базы, чтобы не выводились старые значения

Kyrales committed:
Очистка каталога результатов allure для Yaxunit перед формированием результатов

nixel2007 commented on src/ru/pulsar/jenkins/library/steps/InitInfoBase.groovy:
```diff
@@ -27,6 +28,15 @@ class InitInfoBase implements Serializable {
 
+        def env = steps.env();
+
+        String workspaceAllure = FileUtils.getFilePath("$env.WORKSPACE/build/out/allure").getRemote()
```
вот думаю, может быть в createDir добавить опциональный параметр с очисткой каталога? вызвать внутри что-то вроде
```groovy
if (fileExists(directory)) {
  deleteDir(directory)
}
```

тогда и кописастить не надо будет. и посмотреть, можеть быть и параметр не нужен, а чистить всегда. по идее все, что должно переноситься между стейджами должно быть в стэше

Kyrales replied:
@nixel2007 мне кажется стоит контролируемо очищать. Может быть так, что при следующем запуске (после изменения настроек) и стэш не будет отрабатывать по условиям, а артефакты от него останутся.

nixel2007 replied:
У меня просто во всех сборках стоит очистка воркспейса, поэтому на такого рода проблемы почти не натыкаюсь

Kyrales replied:
> Мне не нравится. Работа с клинапом - это зона ответственности ci.
> 
> Предлагаю решать это средствами дженкинса.

@Segate-ekb ci может очищать это да, но и без очистки все должно быть корректно и нужные хвосты очищаться в виде артефактов от прошлого пайплайна. Библиотека должна работать максимально корректно без особых настроек ci.

Kyrales committed:
fix: Очистка каталогов результатов перед выполнением syntax-check

Kyrales committed:
fix: Очистка каталогов результатов перед выполнением smoke

Kyrales committed:
Merge branch 'firstBitMarksistskaya:develop' into develop

nixel2007 commented:
@Kyrales давай попробуем с флагом в createDir

Kyrales committed:
Merge branch 'develop' of https://github.com/Kyrales/jenkins-lib into develop

Kyrales committed:
Merge branch 'develop' into fix/clear_allure_before

# Conflicts:
#	src/ru/pulsar/jenkins/library/steps/InitInfoBase.groovy

Kyrales committed:
fix: В createDir() Добавлен необязательный параметр deleteDir (по умолчанию false) с проверкой существования и удалением каталога при необходимости

Kyrales commented:
> @Kyrales давай попробуем с флагом в createDir

@nixel2007 да, добавил в createDir() доп.параметр для удаления. Обновил ветку.

Kyrales committed:
fix: Исправление создания каталога

Исправил баг в createDir.groovy (line 1): параметр больше не затеняет Jenkins step deleteDir, а каталог теперь создаётся явно через mkdir на агенте. Это как раз закрывает падение Boolean.call() в SmokeTest и SyntaxCheck.
